### DISCUSSION TITLE
Vision for costs in data dashboards (and alerts)

### DISCUSSION THREAD

**Discussion by kishore-jalleda** [Netflix/dispatch#2432] [General]
Hello, 
I am trying to wrap my head around the costs reported with data dashboards and hoping to get the vision. Specifically, I am trying to understand what the below metrics mean. Instinctively, I am like, incidents have cost associated w/them, but what about the data sources -> costs mapping? Then, when I looked at the code, I see the `cost` attr in the Source class, and it probably means the cost of maintaining that source. I still feel that I am probably missing something. Anyway, here's what I think: 
1. Total Cost (total cost of all the sources?)
2. Avg Cost (avg cost of all the sources?)
3. Top 5 Costs (top 5 sources order by cost desc?) 
4. Top 5 Incidents (top 5 sources order by  len(`incidents` attr) desc limit 5?)

Also, I see model classes defined for alerts, along with methods, api's, etc, and the rel with class source, but don't see a whole lot of it elsewhere in the code Am I missing anything? what's the vision for alerts?   

Thanks,

**Comment by mvilanova** [comment]
Yes, I believe that's the idea -- being able to measure the cost of maintaining data sources that can be used for 1) generating alerts/ events that need to be investigated/ analyzed 2) investigating issues during the handling of incidents or cases. 

The vision for alerts is that Dispatch will monitor the health of the data sources you have onboarded and it will alert you when those are not available (e.g. by performing a SQL query against a table).

Sources, queries, and alerts need more work. @kevgliss will resume his work on them when he's back. He can also provide more information about the vision for them.

**Reply by kishore-jalleda** [reply]
Got it; thanks.
